use move when parameters are passed by value and only copied once

Hailios opened the pull request.

scheibel commented:
Thanks for this PR. I like adding move semantics where applicable.
I see you only add them when passing them to the next function. Wouldn't we have to change signatures to accept rvalue references as well?

Hailios commented:
> Thanks for this PR. I like adding move semantics where applicable.
> I see you only add them when passing them to the next function. Wouldn't we have to change signatures to accept rvalue references as well?

My intent was to utilize move semantics without changing any signatures. Meaning, I just added it where I felt that it was safe for sure.
More use of move semantics can probably be implemented, but would of course require more work as well.

To answer your question, it is not neccessary to add rvalue signatures for the cases I changed. 
What I did was to avoid one copy, and changing the signature to take rvalues could avoid one more copy (the copy to the by-value parameter).
